File: fetch_news.py
import os
import requests
from dotenv import load_dotenv

# โหลดค่าจากไฟล์ .env เข้าสู่ environment variables
load_dotenv()

def get_top_headlines():
    """
    ดึงหัวข้อข่าวเด่นจาก NewsAPI
    """
    # ดึง API Key จาก environment variable
    api_key = os.getenv("NEWS_API_KEY")

    if not api_key:
        print("Error: ไม่พบ NEWS_API_KEY ในไฟล์ .env")
        print("กรุณาสร้างไฟล์ .env และใส่ NEWS_API_KEY=YOUR_KEY_HERE")
        return

    # กำหนด URL ของ API (ตัวอย่าง: ดึงข่าวเด่นจากสหรัฐอเมริกา)
    url = f"https://newsapi.org/v2/top-headlines?country=us&apiKey={api_key}"

    try:
        # ส่ง request ไปยัง API
        response = requests.get(url)
        response.raise_for_status()  # ตรวจสอบว่า request สำเร็จหรือไม่ (status code 200)

        # แปลงข้อมูล JSON ที่ได้กลับมาเป็น Python dictionary
        data = response.json()
        articles = data.get("articles", [])

        if not articles:
            print("ไม่พบข้อมูลข่าวสาร")
        else:
            print("--- Top 10 Headlines from NewsAPI ---")
            for i, article in enumerate(articles[:10]):
                print(f"{i + 1}. {article['title']}")

    except requests.exceptions.RequestException as e:
        print(f"Error fetching news: {e}")

if __name__ == "__main__":
    get_top_headlines()

File: fetch_news.js
const axios = require('axios');

// โหลดค่าจากไฟล์ .env เข้าสู่ process.env
require('dotenv').config();

/**
 * ค้นหาข่าวตาม keyword ที่ระบุจาก NewsAPI
 * @param {string} keyword - คำที่ใช้สำหรับค้นหาข่าว
 */
async function searchNews(keyword) {
  // ตรวจสอบว่ามีการระบุ keyword หรือไม่
  if (!keyword) {
    console.log('กรุณาระบุคำค้นหา (keyword) หลังชื่อไฟล์');
    console.log('ตัวอย่าง: node fetch_news.js "cryptocurrency"');
    return;
  }

  // ดึง API Key จาก environment variable
  const apiKey = process.env.NEWS_API_KEY;

  // ตรวจสอบว่ามี API Key หรือไม่
  if (!apiKey || apiKey === 'YOUR_NEWS_API_KEY_HERE') {
    console.error('Error: ไม่พบ NEWS_API_KEY ในไฟล์ .env');
    console.error('กรุณาสร้างไฟล์ .env และใส่ NEWS_API_KEY ที่ถูกต้อง');
    return;
  }

  // กำหนด URL ของ API สำหรับการค้นหา (everything endpoint)
  const url = `https://newsapi.org/v2/everything`;

  try {
    console.log(`🔍 กำลังค้นหาข่าวเกี่ยวกับ "${keyword}"...`);
    // ส่ง request ไปยัง API ด้วย axios
    const response = await axios.get(url, {
      params: {
        q: keyword, // คำค้นหา
        sortBy: 'publishedAt', // เรียงตามวันที่เผยแพร่ล่าสุด
        language: 'en', // ค้นหาเฉพาะข่าวภาษาอังกฤษ
        apiKey: apiKey,
      },
    });

    // ดึงข้อมูลบทความออกมา (axios จะเก็บข้อมูลไว้ใน property ชื่อ data)
    const articles = response.data.articles || [];

    if (articles.length === 0) {
      console.log(`ไม่พบข่าวที่เกี่ยวกับ "${keyword}"`);
    } else {
      console.log(`--- พบ ${articles.length} ข่าว (แสดง 10 รายการล่าสุด) ---`);
      articles.slice(0, 10).forEach((article, index) => {
        console.log(`${index + 1}. ${article.title}`);
      });
    }
  } catch (error) {
    console.error(`❌ Error fetching news: ${error.message}`);
  }
}

// เรียกใช้ฟังก์ชันหลัก
const keyword = process.argv[2]; // อ่าน argument ตัวแรกที่ส่งเข้ามา
searchNews(keyword);